File: Question_two/most_accessed_pages.js
class PageTracker {
   constructor() {
       this.pageCounts = {};
   };
   accessPage(pageUrl) {
        if (!this.pageCounts[pageUrl]) {
           this.pageCounts[pageUrl] = 0;
        }
        this.pageCounts[pageUrl]++;
    
   };
   getTop5Pages() {
       return Object.entries(this.pageCounts)
           .sort((a, b) => b[1] - a[1])
           .slice(0, 5)
           .map(entry => ({ page: entry[0], count: entry[1] }));
   };
};
const tracker = new PageTracker();
tracker.accessPage('/home');
tracker.accessPage('/about');
tracker.accessPage('/home');
tracker.accessPage('/contact');
tracker.accessPage('/about');
tracker.accessPage('/home');
tracker.accessPage('/delivery');
tracker.accessPage('/signup');
tracker.accessPage('/catalog');
tracker.accessPage('/delivery');
tracker.accessPage('/delivery');
tracker.accessPage('/contact');
tracker.accessPage('/contact');
tracker.accessPage('/contact');
tracker.accessPage('/contact');
tracker.accessPage('/delivery');
console.log(tracker.getTop5Pages());

File: Question_two/most_accessed_pages.kt
class PageTracker {
   private val pageCounts = mutableMapOf<String, Int>()

   fun accessPage(pageUrl: String) {
       pageCounts[pageUrl] = pageCounts.getOrDefault(pageUrl, 0) + 1
    }

   fun getTop5Pages(): List<Pair<String, Int>> {
       return pageCounts.entries
           .sortedByDescending { it.value }
           .take(5)
           .map { it.key to it.value }
   }
}
fun main() {
   val tracker = PageTracker()
   tracker.accessPage("home")
   tracker.accessPage("about")
   tracker.accessPage("home")
   tracker.accessPage("contact")
   tracker.accessPage("about")
   tracker.accessPage("home")

   println(tracker.getTop5Pages())
}
